metashare/api/tests/views.py

```diff
@@ -41,32 +43,220 @@ def test_user_refresh_view(client):
+            get_repo_info.return_value = MagicMock(id=789)
+            response = client.get(reverse("repository-list"))
+
+        assert response.status_code == 200
+        assert response.json() == {
+            "count": 1,
+            "previous": None,
+            "next": None,
+            "results": [
+                {
+                    "id": str(repo.id),
+                    "name": str(repo.name),
+                    "description": "",
+                    "is_managed": False,
+                    "slug": str(repo.slug),
+                    "old_slugs": [],
+                    "repo_url": (
+                        f"https://github.com/{repo.repo_owner}/{repo.repo_name}"
+                    ),
+                }
+            ],
+        }
+
+    def test_get_queryset__bad(
+        self, client, repository_factory, git_hub_repository_factory
+    ):
+        git_hub_repository_factory(
+            user=client.user, repo_id=123, repo_url="https://example.com/test-repo.git"
+        )
+        repo = repository_factory(repo_name="repo", repo_id=123)
+        repository_factory(repo_name="repo2", repo_id=456)
+        repository_factory(repo_name="repo3", repo_id=None)
+        with patch("metashare.api.model_mixins.get_repo_info") as get_repo_info:
+            get_repo_info.side_effect = ResponseError(MagicMock())
+            response = client.get(reverse("repository-list"))
+
+        assert response.status_code == 200
+        assert response.json() == {
+            "count": 1,
+            "previous": None,
+            "next": None,
+            "results": [
+                {
+                    "id": str(repo.id),
+                    "name": str(repo.name),
+                    "description": "",
+                    "is_managed": False,
+                    "slug": str(repo.slug),
+                    "old_slugs": [],
+                    "repo_url": (
+                        f"https://github.com/{repo.repo_owner}/{repo.repo_name}"
+                    ),
+                }
+            ],
+        }
 
-    assert response.status_code == 200
-    assert response.json() == {
-        "count": 1,
-        "previous": None,
-        "next": None,
-        "results": [
-            {
-                "id": str(repo.id),
-                "name": str(repo.name),
-                "description": "",
-                "is_managed": False,
-                "slug": str(repo.slug),
-                "old_slugs": [],
-                "repo_url": f"https://github.com/{repo.repo_owner}/{repo.repo_name}",
-            }
-        ],
-    }
+
+@pytest.mark.django_db
+class TestHookView:
+    def test_202__not_forced(
+        self,
+        settings,
+        client,
+        repository_factory,
+        git_hub_repository_factory,
+        project_factory,
+        task_factory,
+    ):
+        settings.GITHUB_HOOK_SECRET = b""
+        settings.GITHUB_USER_ID = client.user.pk
```

Comment:
@wlonk Should we be using `@override_settings` for this?

Reply:
No, the pytest-django `settings` fixture handles cleanup: https://pytest-django.readthedocs.io/en/latest/helpers.html#settings